src/Accounts/Accounts/Common/EnvironmentCompleterAttribute.cs

```diff
@@ -28,13 +31,16 @@ public static string[] GetEnvironments()
             return profileClient.ListEnvironments(null).Select(x => x.Name).ToArray();
         }
 
-        private static ScriptBlock CreateScriptBlock()
+        /// <summary>
+        /// Implementations CompleteArgument function of the <see cref="IArgumentCompleter"/>.
+        /// </summary>
+        public IEnumerable<CompletionResult> CompleteArgument(string commandName, string parameterName, string wordToComplete, CommandAst commandAst, IDictionary fakeBoundParameters)
         {
-            string script = "param($commandName, $parameterName, $wordToComplete, $commandAst, $fakeBoundParameter)\n" +
-                "$environments = [Microsoft.Azure.Commands.Profile.Common.EnvironmentCompleterAttribute]::GetEnvironments()\n" +
-                "$environments | Where-Object { $_ -Like \"$wordToComplete*\" } | ForEach-Object { [System.Management.Automation.CompletionResult]::new($_, $_, 'ParameterValue', $_) }";
-            ScriptBlock scriptBlock = ScriptBlock.Create(script);
-            return scriptBlock;
+            IEnumerable<string> names = GetEnvironments().Where(env => env.ToLower().StartsWith(wordToComplete.ToLower())) ;
```

Comment:
Have you tested when you doesn't type anything and hit tab, is `wordToComplete` null or empty string? Does the completer work?

Reply:
Tested. Can list word when doesn't type anything or hit tab.